sources/HUBViewModelDiff.m

```diff
@@ -70,7 +79,22 @@ - (BOOL)hasChanges
 {
     return self.insertedBodyComponentIndexPaths.count > 0
         || self.deletedBodyComponentIndexPaths.count > 0
-        || self.reloadedBodyComponentIndexPaths.count > 0;
+        || self.reloadedBodyComponentIndexPaths.count > 0
+        || self.headerComponentIdentifierHasChanged;
+}
+
+- (void)calculateHeaderChangesFromViewModel:(id<HUBViewModel>)fromViewModel toViewModel:(id<HUBViewModel>)toViewModel
+{
+    id<HUBComponentModel> fromHeaderModel = fromViewModel.headerComponentModel;
+    id<HUBComponentModel> toHeaderModel = toViewModel.headerComponentModel;
+
+    if (!fromHeaderModel && !toHeaderModel) {
+        return;
+    }
+
+    if (![fromHeaderModel isEqual:toHeaderModel]) {
+        self.headerComponentIdentifierHasChanged = YES;
```

Comment:
Shouldn't this technically be `headerComponentHasChanged`, as you're comparing the entire model, rather than just the identifier?